operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,290 @@
+package v1alpha1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Kueue is the CRD to represent the Kueue operator.
+// This CRD defines the configuration for Kueue.
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kueue,scope=Cluster
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="Kueue is a singleton, .metadata.name must be 'cluster'"
+type Kueue struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for kueue
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec KueueOperandSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status KueueStatus `json:"status,omitempty"`
+}
+
+type KueueOperandSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// config is the desired configuration
+	// for the Kueue operator.
+	// +required
+	Config KueueConfiguration `json:"config"`
+}
+
+type KueueConfiguration struct {
+	// integrations is a required field that configures the Kueue's workload integrations.
+	// Kueue has both standard integrations, known as job frameworks, and external integrations known as external frameworks.
+	// Kueue will only manage workloads that correspond to the specified integrations.
+	// +required
+	Integrations Integrations `json:"integrations"`
+	// queueLabelPolicy controls how kueue manages workloads
+	// The default behavior of Kueue will manage workloads that have a queue-name label.
+	// Workloads that are missing these label will be ignored by Kueue.
+	// This field is optional.
+	// +optional
+	QueueLabelPolicy *QueueLabelPolicy `json:"queueLabelPolicy,omitempty"`
```

Comment:
To make sure I'm following, the UX of this will look something like:
```yaml
...
queueLabelPolicy:
  policy: QueueNameRequired || QueueNameOptional
  namespaceSelector: ... # potential future field?
...
```

At the very least, I would recommend removing the stuttering in the name. I do still wonder if something like:
```yaml
...
jobManagement:
  labelPolicy: QueueName || None
  namespaceSelector: ... # potential future field?
...
```
would make more sense for what this field is actually intended to represent (i.e configuring how jobs are managed).

Reply:
Hmm. You are following correctly.

Reply:
I like your idea.

Reply:
pushed an update for this.

Reply:
I think I just left a comment saying I didn't like the previous names, looks like you might have fixed that already, sweet!